conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -71,4 +94,94 @@ public static <T> LinkedHashSet<T> newLinkedHashSet(Iterable<? extends T> iterab
         }
         return set;
     }
+
+    public static <T> List<T> newList() {
+        return new ArrayList<>();
+    }
+
+    public static <T> List<T> newList(Iterable<? extends T> iterable) {
+        Preconditions.checkNotNull(iterable, "iterable cannot be null");
+        if (iterable instanceof Collection) {
+            return new ArrayList<>((Collection<T>) iterable);
+        }
+        List<T> list = new ArrayList<>();
+        for (T item : iterable) {
+            list.add(item);
+        }
+        return list;
+    }
+
+    public static <T> List<T> newNonNullList(Iterable<? extends T> iterable) {
+        List<T> arrayList = newList(iterable);
+        for (T item : arrayList) {
+            Preconditions.checkNotNull(item, "iterable cannot contain null elements");
+        }
+
+        return arrayList;
+    }
+
+    public static <T> Set<T> newSet() {
+        return new LinkedHashSet<>();
+    }
+
+    public static <T> Set<T> newSet(Iterable<? extends T> iterable) {
+        Preconditions.checkNotNull(iterable, "iterable cannot be null");
+        if (iterable instanceof Collection) {
+            return new LinkedHashSet<>((Collection<T>) iterable);
+        }
+        Set<T> set = new LinkedHashSet<>();
+        for (T item : iterable) {
+            set.add(item);
+        }
+        return set;
+    }
+
+    public static <T> Set<T> newNonNullSet(Iterable<? extends T> iterable) {
+        Set<T> set = newSet(iterable);
+        for (T item : set) {
+            Preconditions.checkNotNull(item, "iterable cannot contain null elements");
+        }
+
+        return set;
+    }
+
+    // This method returns a list that can't handle nulls. Do not use this unless the nonNullCollections flag is set
+    public static List<Double> newDoubleArrayList() {
```

Comment:
Should be able to just replace these with instantiations of the eclipse-collection implementation

Reply:
Why do we have both `newDoubleArrayList` and `newNonNullDoubleArrayList` neither of which support nulls?

Reply:
My preference is to explicitly declare `NonNull` in the method name. We should not declare the `ArrayList` implementation detail, so something like `public static List<Double> newNonNullDoubleList()` is what I'd go for here

Reply:
fixed bf56b6e08901238154e793c981ea82543431d030

also opted to create `newNonNullSet` and `newNonNullList` to make the code a little simpler (these just instantiate a List, but the add, addAll methods will do the validation...) + can replace these later with a validating collection...